DllImportGenerator/DllImportGenerator/Marshalling/StringMarshaller.Utf16.cs

```diff
@@ -0,0 +1,298 @@
+                                            VariableDeclarator(fixedCharIdentifier)
+                                            .WithInitializer(EqualsValueClause(IdentifierName(managedIdentifier))))),
+                                    Block(
+                                        ExpressionStatement(
+                                            InvocationExpression(
+                                                MemberAccessExpression(
+                                                    SyntaxKind.SimpleMemberAccessExpression,
+                                                    ParseTypeName("System.Buffer"),
+                                                    IdentifierName("MemoryCopy")),
+                                                ArgumentList(
+                                                    SeparatedList(new ArgumentSyntax[] {
+                                                        Argument(IdentifierName(fixedCharIdentifier)),
+                                                        Argument(IdentifierName(stackAllocPtrIdentifier)),
+                                                        Argument(IdentifierName(byteLenIdentifier)),
+                                                        Argument(IdentifierName(byteLenIdentifier))})))))),
+                                // <nativeIdentifier> = <stackAllocIdentifier>;
+                                ExpressionStatement(
+                                    AssignmentExpression(
+                                        SyntaxKind.SimpleAssignmentExpression,
+                                        IdentifierName(nativeIdentifier),
+                                        IdentifierName(stackAllocPtrIdentifier))));
+
+                            // if (<managedIdentifier> == null)
+                            // {
+                            //     <nativeIdentifier> = null;
+                            // }
+                            // else
+                            // {
+                            //     int <byteLenIdentifier> = (<managedIdentifier>.Length + 1) * sizeof(ushort);
+                            //     if (<byteLenIdentifier> > <StackAllocBytesThreshold>)
+                            //     {
+                            //         <nativeIdentifier> = (ushort*)Marshal.StringToCoTaskMemUni(<managedIdentifier>);
+                            //         <usedCoTaskMemIdentifier> = true;
+                            //     }
+                            //     else
+                            //     {
+                            //         ushort* <stackAllocIdentifier> = stackalloc ushort[<managedIdentifier>.Length + 1];
+                            //         fixed (char* <fixedCharIdentifier> = <managedIdentifier>)
+                            //         {
+                            //             Buffer.MemoryCopy(<fixedCharIdentifier>, <stackAllocIdentifier>, <byteLenIdentifier>, <byteLenIdentifier>);
```

Comment:
Instead of assigning the stackalloc to a pointer, pinning the string and using `Buffer.MemoryCopy`, it might be worth assigning the stackalloc to a `Span<char>` or `Span<ushort>` and using `ReadOnlySpan<T>.CopyTo` instead. That way we don't need to pin the string.